src/components/table/schedule/schedule-parts/base-row.component.tsx

```diff
@@ -55,35 +50,39 @@ export function ScheduleRowComponent({
       setSelectedCells([]);
     }
   }, [selectionMode]);
+
   const verboseDates = scheduleLogic?.sections.Metadata?.verboseDates;
 
-  function onContextMenu(cellIndex: number): void {
-    if (scheduleLogic) {
-      scheduleLogic.changeShiftFrozenState(index, cellIndex, setFrozenShifts);
-    }
-  }
+  const onContextMenu = useCallback(
+    (cellIndex: number): void => {
+      if (scheduleLogic) {
+        scheduleLogic.changeShiftFrozenState(index, cellIndex);
+      }
+    },
+    [scheduleLogic, index]
+  );
+
   function saveValue(newValue: string): void {
     if (sectionKey)
       scheduleLogic?.updateRow(
         sectionKey,
         index - 1,
         [...selectedCells, pointerPosition],
-        newValue,
-        (newDataRow) => {
-          setDataRowState(newDataRow);
-          onStateUpdate && onStateUpdate(newDataRow);
-        }
+        newValue
       );
     setSelectedCells([]);
   }
 
-  function isFrozen(cellInd: number): boolean {
-    return (
-      verboseDates?.[cellInd]?.isFrozen ||
-      !!frozenShifts.find((fS) => fS[0] === index && fS[1] === cellInd) ||
-      false
-    );
-  }
+  const isFrozen = useCallback(
+    (cellInd: number): boolean => {
+      return (
+        verboseDates?.[cellInd]?.isFrozen ||
+        // !!frozenShifts.find((fS) => fS[0] === index && fS[1] === cellInd) ||
```

Comment:
Na razie zakomentowałem, ponieważ, `frozenShifts` było zapisane jako lokalny stan, którego się pozbyłem. 
Zostawiłem pro wszelki wypadek, żeby kiedy będziemy implmentować blokowanie zmian, można było zobaczyć jak to było wcześniej. Ale mogę też usunąć